File: blacklist.go
package main

import (
        "html/template"
        "net/http"
        "strconv"
)

func BlacklistHandler(configuration *Configuration, response http.ResponseWriter, request *http.Request) (int, error) {
        response.Header().Set("Content-type", "text/html")

        err := request.ParseForm()
        if err != nil {
                http.Redirect(response, request, "/error", 307)
        }

        params := request.URL.Query()
        var page int
        if len(params["page"]) <= 0 {
                page = 0
        } else {
                page, _ = strconv.Atoi(params["page"][0])
        }

        connection   := connect(configuration.Redis.Host, configuration.Redis.Port)

        blacklist    := connection.Cmd("KEYS", "*:repsheet:ip:blacklisted")
        blacklisted  := Paginate(configuration, blacklist, page, 10)
        templates, _ := template.ParseFiles(
                configuration.TemplateFor("layout"),
                configuration.TemplateFor("pagination"),
                configuration.TemplateFor("blacklist"),
        )
        summary      := Summary{Blacklisted: blacklisted}
        pagination   := GeneratePaginationLinks(blacklist, 10, 0, "/blacklist")
        templates.ExecuteTemplate(response, "layout", Page{Summary: summary, Active: "blacklist", Pagination: pagination})

        return 200, nil
}

File: blacklist.lua
local redis = require 'redis'
local json = require 'json'

function split(str, sep)
    local words = {}
    for w in string.gmatch(str, sep) do 
        table.insert(words, w) 
    end
    return words
end

function blacklist()
    local client = redis.connect('127.0.0.1', 6379)
    local response = client:keys("*:repsheet:ip:blacklisted")

    for k,v in pairs(response) do
        local parts = split(v, "[^:]+")
        response[k] = parts[1]
    end
    
    local t = {}
    t["blacklist"] = response
    
    print(json.encode(t))    
end

blacklist()